### ISSUE TITLE
Windows GPU install compile error

### ISSUE THREAD

**Issue by Whamp** [dmlc/xgboost#3212]
So I am a total noob to cmake and compilers. I've been trying to install this for a week and keep running into problems. I get the error pasted below when running cmake. I don't know if i'm supposed to be appending a path somewhere or if there is a variable i'm supposed to be adding to my environment path but i'm stumped at this point. 

## Environment info
Operating System: Windows 10 Pro

Compiler:MSVS 14 Native tools Command Line and CMAKE 3.11.0

Package used (python/R/jvm/C++): Anaconda 3 64bit

CPU: i7-5820K 

GPU: Nvidia 1080 Ti

Code I ran below: error when running cmake

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC>cd ..\..\

C:\Program Files (x86)>cd ..

C:\>cd users

C:\Users>cd will

C:\Users\Will>cd xgboost
The system cannot find the path specified.

C:\Users\Will>git clone --recursive https://github.com/dmlc/xgboost
Cloning into 'xgboost'...
remote: Counting objects: 22822, done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 22822 (delta 3), reused 7 (delta 2), pack-reused 22798
Receiving objects: 100% (22822/22822), 8.38 MiB | 2.72 MiB/s, done.
Resolving deltas: 100% (13321/13321), done.
Submodule 'cub' (https://github.com/NVlabs/cub) registered for path 'cub'
Submodule 'dmlc-core' (https://github.com/dmlc/dmlc-core) registered for path 'dmlc-core'
Submodule 'nccl' (https://github.com/dmlc/nccl) registered for path 'nccl'
Submodule 'rabit' (https://github.com/dmlc/rabit) registered for path 'rabit'
Cloning into 'C:/Users/Will/xgboost/cub'...
remote: Counting objects: 32642, done.
remote: Total 32642 (delta 0), reused 0 (delta 0), pack-reused 32642
Receiving objects: 100% (32642/32642), 16.46 MiB | 4.79 MiB/s, done.
Resolving deltas: 100% (28621/28621), done.
Cloning into 'C:/Users/Will/xgboost/dmlc-core'...
remote: Counting objects: 4745, done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 4745 (delta 12), reused 13 (delta 6), pack-reused 4713
Receiving objects: 100% (4745/4745), 1.15 MiB | 3.43 MiB/s, done.
Resolving deltas: 100% (2816/2816), done.
Cloning into 'C:/Users/Will/xgboost/nccl'...
remote: Counting objects: 657, done.
remote: Total 657 (delta 0), reused 0 (delta 0), pack-reused 657
Receiving objects: 100% (657/657), 1.38 MiB | 3.28 MiB/s, done.
Resolving deltas: 100% (413/413), done.
Cloning into 'C:/Users/Will/xgboost/rabit'...
remote: Counting objects: 3163, done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 3163 (delta 0), reused 2 (delta 0), pack-reused 3157
Receiving objects: 100% (3163/3163), 906.14 KiB | 3.07 MiB/s, done.
Resolving deltas: 100% (2049/2049), done.
Submodule path 'cub': checked out 'b20808b1b04ec3d6a625e51fbc1eb76f337754ad'
Submodule path 'dmlc-core': checked out 'b5bec5481df86e8e6728d8bd80a61d87ef3b2cd5'
Submodule path 'nccl': checked out 'faeac8333cec3ed7474dc9f1c35088dcd6eea60b'
Submodule path 'rabit': checked out 'a764d45cfb438cc9f15cf47ce586c02ff2c65d0f'

C:\Users\Will>cd xgboost

C:\Users\Will\xgboost>git submodule init

C:\Users\Will\xgboost>git submodule update

C:\Users\Will\xgboost>mkdir build

C:\Users\Will\xgboost>cd build

C:\Users\Will\xgboost\build>cmake .. -G"Visual Studio 14 2015 Win64" -DUSE_CUDA=ON
CMake Error at CMakeLists.txt:2 (project):
  Failed to run MSBuild command:

    MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 4.7.2556.0
    [Microsoft .NET Framework, version 4.0.30319.42000]
    Copyright (C) Microsoft Corporation. All rights reserved.

    Build started 3/30/2018 8:18:38 PM.
    Project "C:\Users\Will\xgboost\build\CMakeFiles\3.11.0\VCTargetsPath.vcxproj" on node 1 (default targets).
    C:\Users\Will\xgboost\build\CMakeFiles\3.11.0\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    Done Building Project "C:\Users\Will\xgboost\build\CMakeFiles\3.11.0\VCTargetsPath.vcxproj" (default targets) -- FAILED.

    Build FAILED.

    "C:\Users\Will\xgboost\build\CMakeFiles\3.11.0\VCTargetsPath.vcxproj" (default target) (1) ->
      C:\Users\Will\xgboost\build\CMakeFiles\3.11.0\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

        0 Warning(s)
        1 Error(s)

    Time Elapsed 00:00:00.33


  Exit code: 1



-- Configuring incomplete, errors occurred!
See also "C:/Users/Will/xgboost/build/CMakeFiles/CMakeOutput.log".

C:\Users\Will\xgboost\build>
```

**Comment by Whamp** [comment]
Solved by switching compiler line from: 
`cmake .. -G"Visual Studio 14 2015 Win64" -DUSE_CUDA=ON`

to:
`cmake .. -G"Visual Studio 15 2017 Win64" -DUSE_CUDA=ON`

**Comment by Whamp** [comment]
when trying to build and release after running this:

`cmake --build . --target xgboost --config Release`

I get this error at the end of the 30 second build:
`"C:\Users\Will\xgboost\build\xgboost.vcxproj" (default target) (1) ->
"C:\Users\Will\xgboost\build\gpuxgboost.vcxproj" (default target) (4) ->
(CustomBuild target) ->
  c:\program files\nvidia gpu computing toolkit\cuda\v9.1\include\crt/host_config.h(135): fatal error C1189: #error:  -
- unsupported Microsoft Visual Studio version! Only the versions 2012, 2013, 2015 and 2017 are supported! [C:\Users\Wil
l\xgboost\build\gpuxgboost.vcxproj]

    80 Warning(s)
    1 Error(s)

Time Elapsed 00:00:31.31

C:\Users\Will\xgboost\build>`

**Comment by Whamp** [comment]
Fixed issue by totally uninstalling visual studio 17, then installing Visual studio 2015. However I got an error when i first ran `cmake .. -G"Visual Studio 14 2015 Win64" -DUSE_CUDA=ON`  saying i didn't have the windows 8.1 sdk kit installed. I fixed this by navigating to: "C:\Users\Will\xgboost\build\CMakeFiles\3.11.0\VCTargetsPath.vcxproj" and attempting to open the file. Before being able to open in Microsoft Visual Studio 2015 community however it notifies me i need to let it download and install certain dependencies including the missing Windows 8.1 SDK. I let it do that and then retried the install and everything worked.
